pkg/ipmi/ipmi_linux.go

```diff
@@ -9,14 +9,17 @@ import (
 	"os"
 )
 
-// Open a channel to an IPMI device /dev/ipmi{devnum}.
-func Open(devnum int) (*IPMI, error) {
-	d := fmt.Sprintf("/dev/ipmi%d", devnum)
-
-	f, err := os.OpenFile(d, os.O_RDWR, 0)
+// OpenPath opens a channel to an IPMI device by path (e.g., /dev/ipmi0).
+func OpenPath(path string) (*IPMI, error) {
+	f, err := os.OpenFile(path, os.O_RDWR, 0)
 	if err != nil {
 		return nil, err
 	}
 
 	return &IPMI{dev: newDev(f)}, nil
 }
+
+// Open opens a channel to an IPMI device by device number (e.g., 0 for /dev/ipmi{devnum}).
+func Open(devnum int) (*IPMI, error) {
```

Comment:
test for this is pretty easy: the test can find a /dev/ipmix that won't open, for some value of x, then call this function to get an error back.

It's kinda silly, save that it ensures expectations continue to be met as other people work on the code.

Also, the common name for things like this is New
ipmi.New("/dev/ipmi%d" etc etc 
and
ipmi.NewNum of NewFromNum or some such. 

Kinda like elf.New, etc.